### PR TITLE
Remove if stream-cancel instruction has been received.

### PR DESCRIPTION
**Pull request by ddragana**
Fixed 1041.

### CODE REVIEW

**Comment by ddragana** [comment]
I started to restructure test a bit, but that is better done in a separate PR.

File: neqo-qpack/src/encoder.rs [hunk 1]
```diff
@@ -1684,4 +1705,33 @@ mod tests {
...
+
+        // send a header block
+        encoder.encode_header_block(
+            1,
+            &[(String::from("content-length"), String::from("1234"))],
+            ENCODE_INDEXED_REF_DYNAMIC,
+            &[],
+        );
+
+        // receive a stream canceled
```
Comment (martinthomson) [inline neqo-qpack/src/encoder.rs:1732]:
```suggestion
        // receive a stream canceled instruction
```

File: neqo-qpack/src/encoder.rs [hunk 2]
```diff
@@ -546,6 +546,41 @@ mod tests {
...
+            let res = self.encoder.send_and_insert(&mut self.conn, header, value);
+            assert!(res.is_ok());
+            self.send_instructions(inst);
+        }
+
+        pub fn encode_header_block(
+            &mut self,
+            stream_id: u64,
+            headers: &[Header],
+            encoded_header_block: &[u8],
```
Comment (martinthomson) [inline neqo-qpack/src/encoder.rs:560]:
```suggestion
            expected_encoding: &[u8],
```

File: neqo-qpack/src/encoder.rs [hunk 3]
```diff
@@ -546,6 +546,41 @@ mod tests {
             // We will try to really change the table only when we send the change capacity instruction.
             self.encoder.send(&mut self.conn)
         }
+
+        pub fn insert(&mut self, header: &[u8], value: &[u8], inst: &[u8]) {
```
Comment (martinthomson) [inline neqo-qpack/src/encoder.rs:550]: Do you want to start using this more?  I see quite a few cases that might benefit from this.

**Review by martinthomson — COMMENTED** [review]
This looks good.  A few opportunities to clean up further though if you feel like it.

File: neqo-qpack/src/encoder.rs (hunk 3 above)
Comment (ddragana) [reply neqo-qpack/src/encoder.rs:550]: Yes. I am planning to use it more, but in a separate PR. This PR is already too big for such a small change.

**Pull request merged**